internal/provider/provider_custom_diffs.go

```diff
@@ -6,13 +6,21 @@ package provider
 import (
 	"context"
 
+	"github.com/hashicorp/go-cty/cty"
 	"github.com/hashicorp/terraform-plugin-framework/path"
 	"github.com/hashicorp/terraform-plugin-framework/resource"
 	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
 	"github.com/hashicorp/terraform-plugin-framework/types"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+type schemaResourceData interface {
```

Comment:
idea borrowed from https://github.com/hashicorp/terraform-provider-aws/blob/3736c100e303f44f81514dbd17805c0a39c02ea7/internal/tags/key_value_tags.go#L707-L711

Reply:
phew 😅 don't need to customizediff, new implementation should be simpler